doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -1824,6 +1824,33 @@ below will fail immediately and won't print anything.
 In any case, the value returned by the fully applied quotation is an
 unspecified dummy Ltac1 closure and should not be further used.
 
+Returning values to Ltac1 from Ltac2 tactics is supported using the `ltac2val` quotation.
+
+.. prodn::
+   ltac_expr += ltac2val : ( @ltac2_expr )
+
+The expression must have type `Ltac1.t`. This quotation does not
```

Comment:
Why this arbitrary restriction? We should be able to add lambda-abstractions just like for ltac2:(...) quotations?

Reply:
Changed
The way `ltac2:(foo |- ...)` works seems hard to use for returning values, but since we return values we can make `ltac2val:(foo |- ...)` be syntactic sugar for `ltac2val:(Ltac1.lambda (fun foo => ...))`

Reply:
AFAICT `ltac2:(foo |- tac)` is like `ltac2val:(Ltac1.lambda (fun foo => tac; ltac1val:(idtac)))`